src/test/java/edu/carroll/cs389application/web/controller/ImageControllerTest.java

```diff
@@ -0,0 +1,113 @@
+package edu.carroll.cs389application.web.controller;
+
+import edu.carroll.cs389application.service.ImageService;
+import edu.carroll.cs389application.service.ImageServiceImpl.ErrorCode;
+import edu.carroll.cs389application.service.UserService;
+import edu.carroll.cs389application.web.form.ImageForm;
+import jakarta.servlet.http.HttpSession;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.mock.web.MockMultipartFile;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.multipart.MultipartFile;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * This class contains unit tests for the ImageController class.
+ */
+public class ImageControllerTest {
+
+    private ImageService imageServiceMock;
+    private UserService userServiceMock;
+    private ImageController controller;
+    private Model modelMock;
+    private HttpSession sessionMock;
+
+    /**
+     * Sets up the test environment by creating mocks and the ImageController instance.
+     */
+    @BeforeEach
+    public void setUp() {
+        imageServiceMock = mock(ImageService.class);
```

Comment:
Good job of setting up your controllers with the necessary mocks for those classes that require injected services!